api/include/opentelemetry/trace/span_context.h

```diff
@@ -81,6 +94,7 @@ class SpanContext final
   trace_api::TraceId trace_id_;
   trace_api::SpanId span_id_;
   trace_api::TraceFlags trace_flags_;
+  trace_api::TraceState trace_state_;
```

Comment:
I'm wondering if we should also consider this new field in `operator==`? Not sure about that either, I'll have a look what other implementations do.

Reply:
Good point. Seems the `operator==` on `SpanContext` is not used anywhere yet, so I just removed it.

Reply:
Added back `operator==` as it is used in the span_context test. Not sure it is necessary here. Added `TODO` in the `operator==` and will take detail look into it.